### PR TITLE
Initial version - review PR

### PR DESCRIPTION
**Pull request by tym2k1**
`psa_util.cpp` is not used in demo scenario but left for now.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
+# TLS Mutual Authentication
+
+This demo covers setting up a TLS-secured server service, generating a
+Certificate Authority (CA), server and client certificates, and demonstrating
+both successful and failed authentication due to certificate validation.
+
+The demo currently doesn't utilize MbedTLS/PSA Crypto API and certificates are generated
+by OpenSSL beforehand
```
Comment (DaniilKl) [inline README.md]:
```suggestion
The demo currently doesn't utilize MbedTLS/PSA Crypto API and certificates are
 generated by OpenSSL beforehand.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+process ensures that the client is talking to the correct server and not an
+impostor. If the server's certificate cannot be verified or is not trusted
+(for example, if it was signed by a different CA), the client will terminate
+the connection attempt.
+
+## Prerequisites
+
+- [Mbed TLS 2.28.8](https://www.trustedfirmware.org/projects/mbed-tls/)
+
+If you have `nix` installed you can easily enter a shell with dependancies with:
```
Comment (DaniilKl) [inline README.md]:
```suggestion
If you have `nix` installed you can easily enter a shell with dependencies with:
```

File: README.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+
+#### Create the CA's self-signed certificate:
+
+Modern versions of TLS and their implementations in libraries
+require the use of Subject Alternative Name (SAN) for hostname
+verification instead of relying solely on the Common Name (CN) field. This
+change improves security and flexibility in certificate handling.
+
+It's recommended to store all certificates in the `keys` directory of repo
```
Comment (DaniilKl) [inline README.md]:
```suggestion
It's recommended to store all certificates in the `keys` directory of repo:
```

File: README.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+```
+
+When developing/testing you can run `make clean` to quickly remove all binaries
+
+```sh
+make clean
+```
+
+Two binaries are produced `mbedtls_script_server` and `mbedtls_script_client`
+which we'll use in testing scenarios
```
Comment (DaniilKl) [inline README.md]:
```suggestion
which we'll use in testing scenarios.
```

File: README.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+### Step 2: Generating the Certificate Authority (CA)
+
+First, create a CA that will sign the server and client certificates.
+
+
```
Comment (DaniilKl) [inline README.md]:
```suggestion

```

File: README.md [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+```sh
+openssl req -new -x509 -days 3650 -config ca.conf -key ca_key.pem -out ca_cert.pem
+```
+
+### Step 3: Generating Server and Client Certificates
+
+#### Generate the server/client certificate signing request (CSR) and sign it:
+
+Repeat the steps for the client certificate, substituting "client" for "server" in the filenames.
+Certificate Configuration (cert.conf)
```
Comment (DaniilKl) [inline README.md]:
```suggestion
Certificate Configuration (cert.conf).
```

**Review by DaniilKl — CHANGES_REQUESTED** [review]
Some typos

File: README.md (hunk 1 above)
Comment (tym2k1) [reply README.md]: [Adresssed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R8)

File: README.md (hunk 2 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R39)

File: README.md (hunk 4 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R71)

File: README.md (hunk 3 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R84)

File: README.md (hunk 5 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L77)

File: README.md (hunk 6 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/402278412049648ec96d9dd5d496085eb6f7b6a0#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R132)

**Comment by tym2k1** [comment]
Review adressed

File: README.md [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+#### Generate the untrusted CA and client certificate
+
+```sh
+$ openssl genrsa -out bad_ca_key.pem 4096
+$ openssl req -new -x509 -days 3650 -config ca.conf -key bad_ca_key.pem -out bad_ca_cert.pem
+$ openssl genrsa -out bad_key.pem 4096
+$ openssl req -new -key bad_key.pem -out bad_csr.pem -config cert.conf
+$ openssl x509 -req -days 365 -in bad_csr.pem -CA bad_ca_cert.pem -CAkey bad_ca_key.pem -CAcreateserial -out bad_cert.pem -extfile cert.conf -extensions req_ext
+```
+#### Correct server certificate incorrect client certificate - access refused
```
Comment (DaniilKl) [inline README.md]:
```suggestion
#### Correct server certificate and incorrect client certificate - access refused
```

File: README.md [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+Loading client private key...
+Setting up SSL configuration...
+Setting up SSL context...
+Connecting to server...
+Starting handshake...
+Handshake failed. Error code: -80
+No verification errors
+```
+
+#### Incorrect server certificate - client doesn't try to connect
```
Comment (DaniilKl) [inline README.md:258]:
```suggestion
#### Incorrect server certificate - client doesn't connect
```

It tries, but fails.

Consider adding explanation to this part, it is not very self-explanatory.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+```sh
+$ openssl genrsa -out bad_ca_key.pem 4096
+$ openssl req -new -x509 -days 3650 -config ca.conf -key bad_ca_key.pem -out bad_ca_cert.pem
+$ openssl genrsa -out bad_key.pem 4096
+$ openssl req -new -key bad_key.pem -out bad_csr.pem -config cert.conf
+$ openssl x509 -req -days 365 -in bad_csr.pem -CA bad_ca_cert.pem -CAkey bad_ca_key.pem -CAcreateserial -out bad_cert.pem -extfile cert.conf -extensions req_ext
+```
+#### Correct server certificate incorrect client certificate - access refused
+
+#### `mbedtls_script_server`
```
Comment (DaniilKl) [inline README.md]:
```suggestion
##### `mbedtls_script_server`
```

**Review by DaniilKl — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (tym2k1) [reply README.md:258]: It verifies that the certificate is not trusted and stops the handshake. It's explained at the [# Client Verifying the Server's Authenticity](https://github.com/3mdeb/MbedTLS_services/tree/review?tab=readme-ov-file#client-verifying-the-servers-authenticity)

File: README.md (hunk 7 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/3c1c4b982826f3947572d469c38f374cc4dd5a68#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R223)

File: README.md (hunk 9 above)
Comment (tym2k1) [reply README.md]: [Adressed](https://github.com/3mdeb/MbedTLS_services/pull/1/commits/3c1c4b982826f3947572d469c38f374cc4dd5a68#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R242) and changed the following headers to match it

**Pull request merged**